enhancements/authentication/login-logout-events.md

```diff
@@ -0,0 +1,159 @@
+---
+title: Login Logout Auditing
+authors:
+  - "@s-urbaniak"
+reviewers:
+  - "@mfojtik"
+  - "@deads2k"
+  - "@stlaz"
+  - "@sttts"
+approvers:
+- "@Anandnatraj"
+- "@oscardoe"
+creation-date: 2021-09-10
+last-updated: 2021-09-10
+status: implementable
+see-also:
+  - "/enhancements/kube-apiserver/audit-policy.md"
+---
+
+# Logging login, login failure and logout events
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement describes capturing login, login failure, and logout events in the OpenShift authentication subsystems.
+
+Currently, login and logout events can only be derived from inspecting audit logs. These include creation or deletion of `oauthaccesstokens` resources as described in the API server [Audit Policy Configuration](https://github.com/openshift/enhancements/blob/master/enhancements/kube-apiserver/audit-policy.md#logging-of-token-with-secure-oauth-storage).
+
+This approach has a couple of drawbacks:
+1. Login events are captured via creation events of `oauthaccesstokens` in audit logs. This includes successful login events only, not login failures.
```

Comment:
note, not necessarily actionable: you might be able to audit certain login failures where the client fails to use the `oauthauthorizetoken` by the token timing out

Reply:
I don't know if it is the same thought as mine, but sounds similar: every unfinished login attempt is a login attempt failure. The reasoning remains unclear.
But the implementation would increase significantly and might be a "non goal"?

Reply:
I don't see how this could be a non-goal. This is already working, it's just a matter of documenting the behavior - `if an oauthauthorizetoken has not been exchanged for an oauthaccesstoken during the lifetime of an oauthauthorizetoken there were likely issues in the login process`. All of the information mentioned is discoverable from the audit logs.

Reply:
You mean we can deduce it.. Yes, this is definitely true.